backend/geneinfo/api/v1.py

```diff
@@ -0,0 +1,78 @@
+import string
+from flask import jsonify, make_response
+import urllib.request
+import json
+import xml.etree.ElementTree as ET
+from backend.geneinfo.config import GeneInfoConfig
+
+
+def geneinfo(geneID: string):
+    # load api key from secrets
+    gene_info_config = GeneInfoConfig()
+    gene_info_config.load()
+    api_key = gene_info_config.ncbi_api_key
+
+    # search for gene UID from ensembl ID
+    if api_key == "test-key":  # adjusting for api key issues in unit tests
+        search_url = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?" f"db=gene&term={geneID}&retmode=json"
+    else:
+        search_url = (
+            "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?"
+            f"db=gene&term={geneID}&api_key={api_key}&retmode=json"
+        )
+    print(search_url)
+    try:
+        search_data = urllib.request.urlopen(search_url).read()
+    except Exception:
+        return make_response(jsonify("Failed search of NCBI database, API key issue"), 404)
+    search_result = json.loads(search_data)
+    if (
+        "esearchresult" not in search_result
+        or "idlist" not in search_result["esearchresult"]
+        or len(search_result["esearchresult"]["idlist"]) < 1
+        or int(search_result["esearchresult"]["count"]) != 1
+    ):
+        return make_response(jsonify("Unexpected NCBI search result"), 404)
```

Comment:
Might be worth logging any unexpected results to simplify debugging in the future